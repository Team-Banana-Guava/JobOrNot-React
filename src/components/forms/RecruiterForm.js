import React, { Component } from 'react';
import SingleInput from './SingleInput';
import Nav from '../navbar/Nav';
import { connect } from 'react-redux';
import { sendSignUp } from '../../actions/auth-actions';

class RecruiterForm extends Component {
    constructor(props) {
        super(props);
        this.state = {
            firstname: '',
            lastname: '',
            username: '',
            email: '',
            password: '',
            company: '',
            jobTitleToFill: '',
            jobCompanyToFill: '',
        };

        this.handleFormSignUp = this.handleFormSignUp.bind(this);
        // this.handleFormUpdate = this.handleFormUpdate.bind(this);
        this.handleFormClear = this.handleFormClear.bind(this);
        this.handleChange = this.handleChange.bind(this);
    }

    handleFormSignUp(e) {
        e.preventDefault();

        const formPayload = {
            firstname: this.state.firstname,
            lastname: this.state.lastname,
            username: this.state.username,
            email: this.state.email,
            password: this.state.password,
            company: this.state.company,
            jobTitleToFill: this.state.jobTitleToFill,
            jobCompanyToFill: this.state.jobCompanyToFill,
        };

        this.props.signUp({ method: 'POST', path: '/signup', body: formPayload });
        this.handleFormClear(e);
    }

    // handleFormUpdate() {
    //     // update form logic goes here
    // }

    handleFormClear(e) {
        e.preventDefault();
        this.setState({
            firstname: '',
            lastname: '',
            username: '',
            email: '',
            password: '',
            company: '',
            jobTitleToFill: '',
            jobCompanyToFill: '',
        });
    }

    handleChange(e) {
        this.setState({ [e.target.name]: e.target.value });
    }
 
    render() {
        return ( 
            <div>
                <Nav />
                <form onSubmit={this.handleFormSignUp}>
                    <h1>THIS FORM SIGNS UP RECRUITERS</h1>
                    <SingleInput 
                        title={'First Name'}
                        name={'firstname'}
                        inputType={'text'}
                        content={this.state.firstname}
                        controlFunc={this.handleChange}
                        placeholder={'First Name'} />
                    <SingleInput 
                        title={'Last Name'}
                        name={'lastname'}
                        inputType={'text'}
                        content={this.state.lastname}
                        controlFunc={this.handleChange}
                        placeholder={'Last Name'} />
                    <SingleInput 
                        title={'User Name'}
                        name={'username'}
                        inputType={'text'}
                        content={this.state.username}
                        controlFunc={this.handleChange}
                        placeholder={'Select a User Name'} />
                    <SingleInput 
                        title={'Email Address'}
                        name={'email'}
                        inputType={'text'}
                        content={this.state.email}
                        controlFunc={this.handleChange}
                        placeholder={'Email Address'} />
                    <SingleInput 
                        title={'Password'}
                        name={'password'}
                        inputType={'text'}
                        content={this.state.password}
                        controlFunc={this.handleChange}
                        placeholder={'Select a Password'} />
                    <SingleInput 
                        title={'Company'}
                        name={'company'}
                        inputType={'text'}
                        content={this.state.company}
                        controlFunc={this.handleChange}
                        placeholder={'Company'} />
                    <SingleInput 
                        title={'Title of Role You are Working to Fill'}
                        name={'jobTitleToFill'}
                        inputType={'text'}
                        content={this.state.jobTitleToFill}
                        controlFunc={this.handleChange}
                        placeholder={'Job Title'} />
                    <SingleInput 
                        title={'Company for Role You are Working to Fill'}
                        name={'jobCompanyToFill'}
                        inputType={'text'}
                        content={this.state.jobCompanyToFill}
                        controlFunc={this.handleChange}
                        placeholder={'Hiring Company'} />
                    <button onClick={this.handleFormClear}>
                        Clear Form
                    </button>
                    <input 
                        type='submit'
                        value='submit'/>
                </form>
            </div>
        );
    };
}

function mapDispatchToProps(dispatch) {
    return {
        signUp: (options) => dispatch(sendSignUp(options))
    };
}

export default connect(null, mapDispatchToProps)(RecruiterForm);

RecruiterForm.propTypes = {
    signUp: React.PropTypes.func
};
