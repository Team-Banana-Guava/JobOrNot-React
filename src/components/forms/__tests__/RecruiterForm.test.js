/* eslint-disable */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import RecruiterForm from '../RecruiterForm';

describe('RecruiterForm', function () {
    it('renders the RecruiterForm form/component', () => {
        const rendered = shallow(<RecruiterForm signedIn={false} signUp={() => {}} />);
        expect(toJson(rendered)).toMatchSnapshot();
    });
});
