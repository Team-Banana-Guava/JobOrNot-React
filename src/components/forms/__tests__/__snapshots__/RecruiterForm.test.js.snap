exports[`RecruiterForm renders the RecruiterForm form/component 1`] = `
<div>
  <Nav
    signedIn={false} />
  <form
    onSubmit={[Function]}>
    <h1>
      THIS FORM SIGNS UP RECRUITERS
    </h1>
    <SingleInput
      content=""
      controlFunc={[Function]}
      inputType="text"
      name="firstName"
      placeholder="First Name"
      title="First Name" />
    <SingleInput
      content=""
      controlFunc={[Function]}
      inputType="text"
      name="lastName"
      placeholder="Last Name"
      title="Last Name" />
    <SingleInput
      content=""
      controlFunc={[Function]}
      inputType="text"
      name="userName"
      placeholder="Select a User Name"
      title="User Name" />
    <SingleInput
      content=""
      controlFunc={[Function]}
      inputType="text"
      name="email"
      placeholder="Email Address"
      title="Email Address" />
    <SingleInput
      content=""
      controlFunc={[Function]}
      inputType="password"
      name="password"
      placeholder="Select a Password"
      title="Password" />
    <SingleInput
      content=""
      controlFunc={[Function]}
      inputType="text"
      name="company"
      placeholder="Company"
      title="Company" />
    <SingleInput
      content=""
      controlFunc={[Function]}
      inputType="text"
      name="jobTitleToFill"
      placeholder="Job Title"
      title="Title of Role You are Working to Fill" />
    <SingleInput
      content=""
      controlFunc={[Function]}
      inputType="text"
      name="jobCompanyToFill"
      placeholder="Hiring Company"
      title="Company for Role You are Working to Fill" />
    <button
      onClick={[Function]}>
      Clear Form
    </button>
    <input
      type="submit"
      value="submit" />
  </form>
</div>
`;
